### PR TITLE
Docs

### LINKED ISSUE #18: [DOC] Add initial package documentation
#### Describe the issue linked to the documentation

The repository currently has minimal documentation.

We need to improve the documentation ahead of the first release by:

- [ ] add detail to README
- [ ]  Create initial sphinx-based documentation

### PR DESCRIPTION
**Pull request by RNKuhns**
#### Reference Issues/PRs

Fixes #18 

#### What does this implement/fix? Explain your changes.

Generates the packages initial documentation, including:

- Creates initial Sphinx based documentation for the package
- Updating README.md to add badges and additional explanation

#### Does your contribution introduce a new dependency? If yes, which one?

No package dependencies are added. Documentation dependencies are specified through optional dependency "docs" in pyproject.toml.

#### What should a reviewer concentrate their feedback on?

- There are really 3 separate things to approve:
   - structure of the Sphinx docs
   - content of the Sphinx docs
   - changes to README.md

#### Any other comments?

None.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by RNKuhns** [comment]
@all-contributors please add @RNKuhns for documentation

**Comment by RNKuhns** [comment]
https://github.com/all-contributors please add @RNKuhns for design

**Comment by fkiraly** [comment]
> https://github.com/all-contributors please add @RNKuhns for design

👀 can we do this in sktime??

**Comment by RNKuhns** [comment]
> > https://github.com/all-contributors please add @RNKuhns for design
> 
> 👀 can we do this in sktime??

@fkiraly you have to enable the bot. Right now I've just got it turned on for sktime/BaseObject. I'm trying to figure out from the docs, whether it will auto-update the CONTRIBUTORS.md too (or whether we need to add the command line tool to our workflow too).

**Comment by RNKuhns** [comment]
@fkiraly I think this is ready for you to look at. I was originally hoping to add text to the user guide, but lets push that to the next release.

Note that the classes in the `baseobject.testing` module are not showing up on the API Ref being built on RTD, but are locally. I'll investigate that tomorrow.

My replies to your earlier review comments are below:

> * governance docs currently copy-paste `sktime` - maybe too early as long as it's an under-the-radar 2-people project? Why not point to `sktime` and only write down the diffs, e.g., different people. And if the project grows, add and/or diverge.

I liked your suggestion about saying we follow sktime's governance generally, but thought it might also be useful have a summary in our docs, so I've summarized the `sktime` governance parts I thought wouldd be most useful. Is that fine with you?

> * re code of conduct committee, there are currently discussions ongoing in `sktime` to make that independent (and different people than) the leadership team - maybe also worth keeping that in mind

I've tried to keep things general. I mainly just wanted to provide a general code of conduct. Totally open to changing any processes on that in the future. Also, it has your email on there currently for CoC contact (since you were one of contacts on sktime). Are you fine with that?

> Perhaps it's worth splitting the governance and the tech docs? I have no major objection/questions regarding the tech docs.

I like the idea of having some basic governance to start with. Let me know if your fine with the updated governance section. Otherwise we can refine before the release.

> tech docs:
> * history - maybe worth pointing out that `BaseObject` evolved first in `sklearn` (as `BaseEstimator`) and then as `BaseObject` in `sktime`?

Good catch, I've added a blurb about this. Let me know if that works.

> * structure - is it worth having user documentation and tutorials as separate? in `sktime` we had the problem that there was a multiplication of locations with introductory materials, and some was forever a construction site, until we converged it into fewer locations.

I am a fan of having them separate. I think that eventually the user guide should be mostly textual. We'll describe the concepts around the different interfaces. I see the tutorials as having all the "how do you use this" examples.

**Comment by RNKuhns** [comment]
@fkiraly I've fixed the outstanding error about the API Reference not showing the testing module (needed to add pytest to package dependencies, since it is used there).

File: docs/source/user_documentation/installation.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+.. note::
+
+    We are still working on creating releases of ``baseobject`` on ``conda``.
+    If you would like to help, please open a pull request.
+
+Release versions - troubleshooting
+----------------------------------
+
+Missing soft fependencies
```
Comment (fkiraly) [inline docs/source/user_documentation/installation.rst:51]: typo

**Review by fkiraly — APPROVED** [review]
Good for v0.1.0

I have some comments which I will collect in an issue instead

**Pull request merged**